### PR TITLE
Support local development in the Magento 2 image

### PR DESCRIPTION
**Pull request by kierenevans**
A few changes to better support local development with the Magento 2 image.

We'll be updating the https://github.com/continuouspipe/demo-magento2 with the example local development setup a bit later on.

### CODE REVIEW

**Comment by andytson-inviqa** [comment]
@kierenevans have you tried docker-remotefs yet (bound to be the fastest method). It would possibly mean best to not ignore (tho rsync could sync everything instead)

**Comment by andytson-inviqa** [comment]
it'd also be good later to port the M2 s3 asset stuff to what the parent images do

File: magento2-nginx/fpm-7.0/usr/local/share/magento2/magento_functions.sh [hunk 1]
```diff
@@ -381,7 +385,41 @@ function do_magento_tail_logs() {
...
+      --admin-password='${MAGENTO_ADMIN_PASSWORD}' \
+      --admin-email='admin@example.com' \
+      --admin-firstname='Development' \
+      --admin-lastname='Admin'"
+    set -x
+  fi
+}
+
 function do_magento2_build() {
+  if [ ! -f '/app/composer.json' ]; then
```
Comment (andytson-inviqa) [inline magento2-nginx/fpm-7.0/usr/local/share/magento2/magento_functions.sh]: what's the significance of there not being a composer.json? A comment would be best if significant reason to

**Comment by kierenevans** [comment]
Completely forgot that I had another branch for testing docker-remotefs. Let me apply those changes here too.

**Comment by kierenevans** [comment]
Changes applied, they work for me locally!

**Comment by sroze** [comment]
Looks good to me 👍

**Comment by sroze** [comment]
👏 @kierenevans

**Pull request merged**